docs/Java-SpringBoot-Developer.md

```diff
@@ -0,0 +1,369 @@
+
+
+### JAVA-SPRINGBOOT-COSMOSDB-SDK-GAP : support for usage GetRequestCharge​ metrics
+
+GetRequestCharge metrics are not supported in spring-boot-cosmosdb sdk.
+
+
+### JAVA-SPRINGBOOT-COSMOSDB-SDK-GAP : No native query support with @Query Annotation 
+
+ ```java
+ @Repository
+public interface MoviesRepository extends DocumentDbRepository<Movie, String>  {
+    List<Movie> findByMovieId(String movieId);
+    List<Movie> findByTextSearchContaining(String movieName);
+    //Native queries example commented below are not supported by spring-boot-cosmosdb sdk
+    //@Query(“Select * from c where c.year= :1 order by order by c.year ”)
+    //List<Movie> findMoviesByYear(int year);
+
+}
+ ```
+
+### Solution:
+1. Customers are moving to Cosmos Mongo for easier developer experience with features on Native queries, Cross doc queries, paginations, flexibility support
+
+## AKS Pod Identity Support
+
+WIP 
+
+
+## Versioning
+
+Helium dynamically builds a version string based spring-boot snapshot version and date time of build. This is displayed in both the Healthz output as well as the Swagger UI. 
+
+[BuildConfig.java](BuildConfig)
+
+
+```xml
+    <plugin>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>build-info</id>
+            <goals>
+              <goal>build-info</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      
+```
+
+```java
+
+public class BuildConfig {
+
+    @Autowired
+    BuildProperties buildProperties;
+
+    public String getBuildVersion(){
+
+        String buildName = buildProperties.getName();
+        String buildVersion = buildProperties.getVersion();
+        String buildTime =  String.valueOf(buildProperties.getTime().getEpochSecond());
+        System.out.println("buildName " + buildName +"_"+ "buildVersion "+ buildVersion + "_" + "buildTime "+ buildTime );
+        return buildVersion+"."+buildTime ;
+    }
+}
+
+```
+
+
+### Application Insights
+
+Get an Application Insights instrumentation key by creating an Application Insights resource. Set the application type to Java web application.
+
+Store the application insights instrumentation key in Key Vault to configure Helium to use Application Insights. When configured, the DI creates a singleton instance of TelemetryClient that can be used to track custom events and metrics.
+
+#### Adding Application Insights from Spring framework DI 
+
+Add the dependency "azure-keyvault-secrets-spring-boot-starter" and "azure-client-authentication" to the maven POM file
+
+[POM.xml] (https://github.com/microsoft/helium-java/blob/master/pom.xml) 
+
```

Comment:
Remove space between [POM.xml] and (https...) so the link renders correctly. 
I.e. : [POM.xml](https://github.com/microsoft/helium-java/blob/master/pom.xml)